### ISSUE TITLE
How do you handle multipel developers

### ISSUE THREAD

**Issue by mrwellmann** [LotteMakesStuff/LMS.Version#1]
Hi Lotte,

the asset looks great and thanks for sharing!

I was thinking of building something very basic to auto increase the build number for our project.
The issue I have not solved until now is how to handle multiple developers doing test builds in parallel. The build number will increase independently for each developer. As we develop features in parallel often when merge back into our develop branch in git there will be a conflict.

Do you have any neat solution for this problem or did you just live with the conflicts?

**Comment by LotteMakesStuff** [comment]
This hasn't been an issue for the projects I've used LMS.Version on, as the teams I work with tend to be small and one person usually manages the building on all platforms, but yeah this could be a problem for larger teams.

As this versioning tool keeps version information in a standard unity asset, that data can be tracked and versioned using whatever version control you use for your project. So, whenever were doing builds we make sure to pull before a build and commit after. that helps to ensure version consistency. That wont help if your building from multiple branches tho.

Other than that, I think its worth pointing out that its okay if version numbers are different across different platforms, as long a version number always goes up on each individual platform, if that makes sense. Out of the box, LMS.Version deals with version numbers made of three parts, a Major & Minor number, and a build number.

For us we see the Major.Minor number as the thing that really identifies a feature level in the game. When doing a multi-platform release, that's the bit we care about being the same, and its the bit that a consumer of your game would most care about. These numbers have to be manually set on the version asset. 

That third number we call the Build number and *really* we don't care what it is or if it's different across different platforms. The only thing we do care about is that I goes up over time! Actually, we kind of want it to be different on different platforms! Each build on each platform is a totally different build! In LMS.Version this is the only part of the version number that actually auto increments. So, when its time to start building for a milestone or a patch, we collectively decide what the new Major.Minor version number is, we set that on the version asset and then clear the build number back to zero, and make sure that change is merged into each platform branch on git. then we just build as needed. 

 This is also totally an area where you can modify the tool to meet your development philosophy.

**Comment by mrwellmann** [comment]
Thanks for the detailed Answer!
